src/libs/docker/commands.js

```diff
@@ -124,41 +134,118 @@ const login = async ({ providerUrl, user, token }) => {
+        log && Logger.stderr(data)
+        output.error.push(data)
+      },
+      onExit: (exitCode) => (
+        res({
+          data: output.data.join(''),
+          error: output.error.join(''),
+          exitCode,
+        })
+      )
+    })
+  })
+}
+
 /**
  * Pushes a local docker image to the docker provider base on the url
  * @function
- * @param {string} url - Url to push the image to
+ * @param {string|Object} url - Url to push the image to
+ * @param {boolean} log - Log messages and docker commands
+ * @param {boolean} skipError - Skip throwing an error if command fails
  *
- * @returns {void}
+ * @returns {boolean} True if the image could be pushed
  */
-const push = async url => {
+const push = async (url, log, skipError) => {
+  const toPush = isStr(url)  ? { url, log, skipError } : url
 
-  Logger.spacedMsg(`  Pushing docker image to url`, url)
+  toPush.log && Logger.spacedMsg(`Pushing docker image to`, toPush.url)
 
-  const { error, data } = await spawnProc(`docker push ${ url }`)
+  const exitCode = await spawnProc(`docker push ${toPush.url}`)
 
-  return error && !data
-    ? apiError(error)
-    : Logger.success(`  Finished pushing Docker image to provider!`)
+  if(exitCode)
+    return toPush.skipError ? false : apiError(`docker push ${toPush.url}`)
+
+  toPush.log && Logger.success(`\nFinished pushing ${toPush.url} to provider!\n`)
+
+  return exitCode
 }
 
 /**
  * Pulls a docker image from a provider to the local machine
  * @function
- * @param {string} url - Url to pull the image from
+ * @param {string|Object} url - Url to pull the image from
+ * @param {boolean} log - Log messages and docker commands
+ * @param {boolean} skipError - Skip throwing an error if command fails
  *
- * @returns {void}
+ * @returns {boolean} True if the image could be pulled
  */
-const pull = async url => {
+const pull = async ({ url, log=true, skipError=false, pipe=false }) => {
+  const toPull = isStr(url) ? { url, log, skipError } : url
 
-  Logger.spacedMsg(`  Pulling docker image from url`, url)
+  toPull.log && Logger.spacedMsg(`Pulling docker image from`, toPull.url)
 
-  const { error, data } = await spawnProc(`docker pull ${ url }`)
+  if(pipe){
+    const { error, data, exitCode } = await dockerCLiPipe(
+      `docker pull ${toPull.url}`,
+      { loading: { title: `- Downloading Image`, offMatch: [ `Status:` ] }},
+      { filter: [toPull.url], log }
+    )
 
-  return error && !data
-    ? apiError(error)
-    : Logger.success(`  Finished pulling Docker image from provider!`)
+    if(error.length || exitCode)
+      return toPull.skipError
+        ? { data, error, exitCode }
+        : apiError(error)
+
+    toPull.log && Logger.success(`\nFinished pulling ${toPull.url} from provider!\n`)
+    return { data, error, exitCode }
+
+  }
+  else {
+    const exitCode = await spawnProc(`docker pull ${toPull.url}`)
+    if(exitCode)
+      return toPush.skipError ? false : apiError(`docker push ${toPull.url}`)
```

Comment:
I ran into an error here. when running `keg herkin pack run herkin:zen-463-image-from-updates`

looks like it's just a typo

![image](https://user-images.githubusercontent.com/3317835/106797864-68e5b700-661a-11eb-9c20-de068bffbe99.png)